### PR TITLE
[Merged by Bors] - chore(Data.Finset): un-`@[simp]` `filter_{true,false}_of_mem`

### PR DESCRIPTION
**Pull request by Vierkantor**
These are okay `@[simp]` lemmas in principle but sometimes discharging the hypothesis is very expensive.

This PR adds some extra `@[simp]` lemmas instead; those should reduce the impact of the change quite a lot.

Zulip thread: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Near-timeout.20in.20.60mulEnergy_card_filter.60

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Basic.lean [hunk 1]
```diff
@@ -2645,15 +2645,21 @@ theorem filter_nonempty_iff : (s.filter p).Nonempty ↔ ∃ a ∈ s, p a := by
...
 theorem filter_false_of_mem (h : ∀ x ∈ s, ¬p x) : s.filter p = ∅ := filter_eq_empty_iff.2 h
 #align finset.filter_false_of_mem Finset.filter_false_of_mem
 
+-- Specialization of `filter_true_of_mem` that is much faster to apply by `simp`
+@[simp]
+theorem filter_true : s.filter (fun _ => True) = s := filter_true_of_mem (by aesop)
+
+-- Specialization of `filter_false_of_mem` that is much faster to apply by `simp`
+@[simp]
+theorem filter_false : s.filter (fun _ => False) = ∅ := filter_false_of_mem (by aesop)
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Basic.lean:2661]: I am *pretty sure* those lemmas already exist. Mind checking?

**Comment by YaelDillies** [comment]
Two of those files are mine 😳

File: Mathlib/Data/Finset/Basic.lean (hunk 1 above)
Comment (Vierkantor) [reply Mathlib/Data/Finset/Basic.lean:2661]: Indeed these do exist; with different capitalization. Will replace accordingly.

File: Mathlib/Data/Finset/Basic.lean [hunk 2]
```diff
@@ -2665,6 +2665,7 @@ theorem filter_congr {s : Finset α} (H : ∀ x ∈ s, p x ↔ q x) : filter p s
 
 variable (p q)
 
+@[simp]
```
Comment (mattrobball) [inline Mathlib/Data/Finset/Basic.lean:2668]:
```suggestion
```
It looks like this is solved `by simp`. It would nice to be able to trust the linter.

**Review by mattrobball — APPROVED** [review]

**Comment by mattrobball** [comment]
Surprising this makes `simp` almost 2% better!

**Comment by jcommelin** [comment]
:chart_with_downwards_trend: :octopus: :tada: 

bors merge

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**